core/src/main/scala/org/apache/spark/scheduler/StageInfo.scala

```diff
@@ -36,7 +36,7 @@ class StageInfo(
     val rddInfos: Seq[RDDInfo],
     val parentIds: Seq[Int],
     val details: String,
-    val internalAccumulators: Seq[Accumulator[_]] = Seq.empty,
+    val taskMetrics: TaskMetrics = null,
```

Comment:
if that's the case, it might be better to always create a taskmetric rather than leave it at null